### PR TITLE
Ssh task class

### PR DESCRIPTION
**Pull request by antonrohr**
Added `ReadStdErr()` to Channel.

Refactored `ExecChannelManager` into `Task<T>`

### CODE REVIEW

File: OrbitSsh/include/OrbitSsh/Task.h [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+  outcome::result<void> Read();
+  std::optional<ResultType> result_ = std::nullopt;
+  Session* session_ptr_;
+  std::optional<Channel> channel_;
+  std::string command_;
+  std::function<void(std::string, std::optional<ResultType>*)>
+      std_out_callback_;
+  std::function<void(std::string, std::optional<ResultType>*)>
+      std_err_callback_;
+  std::function<ResultType(int, std::optional<ResultType>*)> exit_callback_;
```
Comment (beckerhe) [inline OrbitSsh/include/OrbitSsh/Task.h:49]: Can you add type aliases with `using` for these std::functions? Then they will get a lot shorter.

File: OrbitSsh/include/OrbitSsh/Channel.h [hunk 2]
```diff
@@ -28,6 +28,7 @@ class Channel {
   static outcome::result<Channel> OpenChannel(Session* session_ptr);
 
   outcome::result<std::string> Read(int buffer_size = 0x400);
+  outcome::result<std::string> ReadStdErr(int buffer_size = 0x400);
```
Comment (beckerhe) [inline OrbitSsh/include/OrbitSsh/Channel.h:31]: Can you rename Read to ReadStdOut?

File: OrbitSsh/include/OrbitSsh/Task.h [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+outcome::result<void> Task<ResultType>::Read() {
+  if (std_out_callback_) {
+    auto std_out = channel_->Read();
+
+    // if std out reading has error -> return it
+    if (!std_out && !shouldITryAgain(std_out)) return std_out.error();
+
+    // if std out not empty -> std out callback
+    if (std_out && !std_out.value().empty()) {
+      std_out_callback_(std_out.value(), &result_);
```
Comment (beckerhe) [inline OrbitSsh/include/OrbitSsh/Task.h:80]: Can you move the string into the callback?

File: OrbitSsh/include/OrbitSsh/Task.h [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+  if (std_err_callback_) {
+    auto std_err = channel_->ReadStdErr();
+
+    // if std err reading has error -> return it
+    if (!std_err && !shouldITryAgain(std_err)) return std_err.error();
+
+    // if std err not empty -> std err callback
+    if (std_err && !std_err.value().empty()) {
+      std_err_callback_(std_err.value(), &result_);
```
Comment (beckerhe) [inline OrbitSsh/include/OrbitSsh/Task.h:92]: Again move!

File: OrbitSsh/include/OrbitSsh/Task.h [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+      state_ = State::kFinished;
+      return outcome::success(
+          exit_callback_(channel_->GetExitStatus(), &result_));
+    }
+    case State::kFinished:
+      return Error::kTaskUsedAfterFinish;
+      break;
+  }
+
+  return outcome::success();
```
Comment (beckerhe) [inline OrbitSsh/include/OrbitSsh/Task.h:129]: This makes no sense anymore, right? Error::kTaskUsedAfterFinish should trigger here, right?

Comment (antonrohr) [reply OrbitSsh/include/OrbitSsh/Task.h]: true, done

File: OrbitSsh/include/OrbitSsh/Task.h (hunk 4 above)
Comment (antonrohr) [reply OrbitSsh/include/OrbitSsh/Task.h]: done

File: OrbitSsh/include/OrbitSsh/Task.h (hunk 3 above)
Comment (antonrohr) [reply OrbitSsh/include/OrbitSsh/Task.h]: done

File: OrbitSsh/include/OrbitSsh/Task.h (hunk 1 above)
Comment (antonrohr) [reply OrbitSsh/include/OrbitSsh/Task.h:49]: done

File: OrbitSsh/include/OrbitSsh/Channel.h (hunk 2 above)
Comment (antonrohr) [reply OrbitSsh/include/OrbitSsh/Channel.h:31]: done

**Pull request merged**